### PR TITLE
12 jun develop

### PR DESCRIPTION
**Pull request by samreenzaidi**

### CODE REVIEW

**Comment by samreenzaidi** [comment]
ok

File: app.js [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+// options is optional 
+glob("./server/**/*.json", null, function(er, files) {
+    if (er)
+        console.log(er);
+    else {
+        jsonConcat({
+            src: files,
+            dest: "./server/db.json"
+        }, function(json) {
+            console.log(json);
```
Comment (abhirajambasta) [inline app.js:19]: is it needed?

File: app.js [hunk 2]
```diff
@@ -0,0 +1,59 @@
+var glob = require("glob"),
```
Comment (abhirajambasta) [inline app.js:1]: remove commented codes

File: resources/css/cart.css [hunk 3]
```diff
@@ -0,0 +1,73 @@
+.cart {
+  display: none;
```
Comment (abhirajambasta) [inline resources/css/cart.css]: why an element with display: none has so many other properties

File: resources/js/index.js [hunk 4]
```diff
@@ -0,0 +1,31 @@
+var express = require('express'),
```
Comment (abhirajambasta) [inline resources/js/index.js:1]: need to have error handling for all these request

File: resources/js/main.js [hunk 5]
```diff
@@ -0,0 +1,7 @@
+function login(){
```
Comment (abhirajambasta) [inline resources/js/main.js]: why this again?

File: resources/js/product.js [hunk 6]
```diff
@@ -0,0 +1,31 @@
+var express = require('express'),
```
Comment (abhirajambasta) [inline resources/js/product.js:1]: same as above

File: resources/js/script.js [hunk 7]
```diff
@@ -0,0 +1,49 @@
+/*-------------------------- slider functonality --------------------------*/
+var slideIndex = 1;
+const showSlides = (n) => {
+    var i;
+    var slides = document.getElementsByClassName("mySlides");
+    var dots = document.getElementsByClassName("dot");
+    if (n > slides.length) { slideIndex = 1 }
+    if (n < 1) { slideIndex = slides.length }
+    for (i = 0; i < slides.length; i++) {
```
Comment (abhirajambasta) [inline resources/js/script.js:9]: avoid taking length in loop body

File: resources/js/script.js [hunk 8]
```diff
@@ -0,0 +1,49 @@
...
+    showSlides(slideIndex += n);
+};
+
+currentSlide = (n) => {
+    showSlides(slideIndex = n);
+};
+
+
+/*-------------------------- Cart functonality --------------------------*/
+openCart = () => {
```
Comment (abhirajambasta) [inline resources/js/script.js]: ideally toggling class on the cart would have done in a single line

File: resources/scss/index.scss [hunk 9]
```diff
@@ -0,0 +1,125 @@
+@mixin bottom_style {
+    margin-bottom: 20px;
+    &::after {
+        content: "";
+        display: block;
+        border: 1px solid #cdcdcd6e;
```
Comment (abhirajambasta) [inline resources/scss/index.scss:6]: variables?

File: resources/scss/index.scss [hunk 10]
```diff
@@ -0,0 +1,125 @@
+@mixin bottom_style {
+    margin-bottom: 20px;
+    &::after {
+        content: "";
+        display: block;
+        border: 1px solid #cdcdcd6e;
+        margin: 0 30%;
+        box-shadow: 0 2px 2px #cdcdcd6e;
```
Comment (abhirajambasta) [inline resources/scss/index.scss:8]: same as above

File: resources/scss/index.scss [hunk 11]
```diff
@@ -0,0 +1,125 @@
...
+    }
+}
+
+.prod_image_section {
+    width: 30%;
+}
+
+.prod_section {
+    .left {
+        float: left;
```
Comment (abhirajambasta) [inline resources/scss/index.scss:36]: generalised classes, should be in global css

File: resources/scss/index.scss [hunk 12]
```diff
@@ -0,0 +1,125 @@
...
+}
+
+.home_container {
+    padding: 0 40px;
+}
+
+.mySlides {
+    display: none;
+    img {
+        text-align: center;
```
Comment (abhirajambasta) [inline resources/scss/index.scss:56]: is this required with parent hidden?

File: resources/scss/index.scss [hunk 13]
```diff
@@ -0,0 +1,125 @@
...
+    margin: 0 2px;
+    background-color: #bbb;
+    border-radius: 50%;
+    display: inline-block;
+    transition: background-color 0.6s ease;
+}
+
+.active,
+.dot:hover {
+    background-color: #717171;
```
Comment (abhirajambasta) [inline resources/scss/index.scss:106]: variables?

File: resources/scss/main.scss [hunk 14]
```diff
@@ -0,0 +1,181 @@
...
+    font-weight: 300;
+    text-decoration: none;
+    color: black;
+}
+
+.page_nav_link {
+    &:link {
+        @include page_nav_link_style;
+    }
+    &:visited {
```
Comment (abhirajambasta) [inline resources/scss/main.scss]: can this and the upper one be declared in a single line?

File: resources/scss/main.scss [hunk 15]
```diff
@@ -0,0 +1,181 @@
...
+        @include page_nav_link_style;
+    }
+}
+
+.header_wrapper {
+    padding: 2px 40px;
+    border-bottom: 2px solid #33333314;
```
Comment (abhirajambasta) [inline resources/scss/main.scss]: variables?

File: resources/scss/main.scss [hunk 16]
```diff
@@ -0,0 +1,181 @@
+$font-family: 'Dosis',
```
Comment (abhirajambasta) [inline resources/scss/main.scss]: follow a constant class declaration pattern, either 'hyphen-separated', 'underscore_separated' or 'camelCase'. Don't mix them.

File: resources/scss/main.scss [hunk 17]
```diff
@@ -0,0 +1,181 @@
...
+
+a {
+    text-decoration: none;
+}
+
+@mixin page_nav_link_style {
+    display: inline-block;
+    font-weight: 300;
+    text-decoration: none;
+    color: black;
```
Comment (abhirajambasta) [inline resources/scss/main.scss]: variables?

File: resources/scss/main.scss [hunk 18]
```diff
@@ -0,0 +1,181 @@
...
+
+input[type=text],
+input[type=email],
+input[type=password] {
+    @include input;
+    &:focus+label {
+        position: absolute;
+        opacity: 1;
+        display: block;
+        color: #2FB9C6;
```
Comment (abhirajambasta) [inline resources/scss/main.scss]: same as above

File: resources/scss/main.scss [hunk 19]
```diff
@@ -0,0 +1,181 @@
...
+}
+
+.register_sect {
+    padding: 30px;
+}
+
+.btn {
+    color: #fff;
+    border: #BF2957;
+    background-color: #BF2957;
```
Comment (abhirajambasta) [inline resources/scss/main.scss]: same as above

File: resources/scss/product.scss [hunk 20]
```diff
@@ -0,0 +1,170 @@
...
+    float: left;
+    .navbar-content {
+        a {
+            padding: 6px 8px 6px 16px;
+            display: block;
+            color: #000;
+            &::after {
+                content: "";
+                display: block;
+                border: 1px solid #1d151533;
```
Comment (abhirajambasta) [inline resources/scss/product.scss]: same as above

File: resources/scss/product.scss [hunk 21]
```diff
@@ -0,0 +1,170 @@
...
+    h6 {
+        background: #EAEAEA;
+        margin: 5px 0px;
+        padding: 5px;
+        min-height: 110px;
+        color: #333333cf;
+    }
+    .desktop_prod {
+        display: block;
+        margin: 15px 0px 25px 0px;
```
Comment (abhirajambasta) [inline resources/scss/product.scss]: use shorthand

File: resources/scss/product.scss [hunk 22]
```diff
@@ -0,0 +1,170 @@
...
+    }
+}
+
+
+/* 
+  ##Device = Most of the Smartphones Mobiles (Portrait)
+  ##Screen = B/w 320px to 479px
+*/
+
+@media (min-width: 320px) and (max-width: 480px) {
```
Comment (abhirajambasta) [inline resources/scss/product.scss:107]: mobile first approach should be followed throughout the project.

File: views/index.hbs [hunk 23]
```diff
@@ -0,0 +1,32 @@
+<main class="home_container">
```
Comment (abhirajambasta) [inline views/index.hbs:1]: indentation

File: views/login.hbs [hunk 24]
```diff
@@ -0,0 +1,20 @@
+<main class="form_content">
+    <aside class="left_content">
+        <h1>Login</h1>
+        <p>Get access to your Orders, Wishlist and Recommendations</p>
+    </aside>
+    <section class="right_content">
+        <form id="login_form">
```
Comment (abhirajambasta) [inline views/login.hbs:7]: form method, action etc are missing

File: views/login.hbs [hunk 25]
```diff
@@ -0,0 +1,20 @@
+<main class="form_content">
+    <aside class="left_content">
+        <h1>Login</h1>
+        <p>Get access to your Orders, Wishlist and Recommendations</p>
+    </aside>
+    <section class="right_content">
+        <form id="login_form">
+            <input type="email" id="login_email" name="email" placeholder="Email" onfocus="this.placeholder = ''"
```
Comment (abhirajambasta) [inline views/login.hbs:8]: how are these validated on front-end

File: views/partials/cart.hbs [hunk 26]
```diff
@@ -0,0 +1,34 @@
...
+                        <span><i class="fa fa-plus" aria-hidden="true"></i></span>
+                        <span><i class="fa fa-times" aria-hidden="true"></i></span>
+                        <span>Rs.187</span>
+                        <span class="total_amt">Rs.187</span>
+                    </div>
+                </div>
+            </div>
+            <div class="cart-caption">
+                <img src="/images/lowest-price.png" alt="" width="30%" />
+                <p>You won't find it cheaper anywhere</p>
```
Comment (abhirajambasta) [inline views/partials/cart.hbs:26]: can't we get these from json?

File: views/partials/footer.hbs [hunk 27]
```diff
@@ -0,0 +1,3 @@
+<footer>
+    Copyright &copy; 2011-2019 Sabka Bazaar Grocery Supplies Pvt Ltd
```
Comment (abhirajambasta) [inline views/partials/footer.hbs:2]: still the date is not dynamic

**Pull request closed without merging**